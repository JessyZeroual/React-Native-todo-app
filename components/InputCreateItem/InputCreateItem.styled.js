import styled from "styled-components/native";
import { primaryColor } from "../../constant/style-constants";

export const Form = styled.View`
  display: flex;
  align-items: center;
  flex-direction: row;
  background: #fff;
  height: 40px;
  border-radius: 40px;
  margin: 10px 20px;
  margin-top: -20px;
  padding-left: 20px;
  padding-right: 10px;
`;

export const Input = styled.TextInput`
  flex: 1;
`;

export const ButtonSubmit = styled.TouchableHighlight`
  display: flex;
  justify-content: center;
  align-items: center;
  height: 30px;
  width: 30px;
  background-color: ${primaryColor};
  border-radius: 30px;
`;
