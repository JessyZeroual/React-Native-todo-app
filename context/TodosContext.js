import React from "react";

const TodosContext = React.createContext({});

export const TodosContextProvider = TodosContext.Provider;
export const TodosContextConsumer = TodosContext.Consumer;
export default TodosContext;
