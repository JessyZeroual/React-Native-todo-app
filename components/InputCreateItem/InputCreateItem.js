import React, { useState, useContext } from "react";
import TodosContext from "../../context/TodosContext";
import Icon from "react-native-vector-icons/MaterialIcons";
import {
  InputCreateItemWrapper,
  Shape,
  Form,
  Input,
  ButtonSubmit,
} from "./InputCreateItem.styled";

const InputCreateItem = () => {
  const [text, setText] = useState("");
  const todosContext = useContext(TodosContext);
  const { _postTodo } = todosContext.dispatch;
  return (
    <InputCreateItemWrapper>
      <Shape />
      <Form style={shadow}>
        <Input
          onChangeText={(text) => setText(text)}
          placeholder="Add item"
          placeholderTextColor="grey"
          value={text}
          onSubmitEditing={() => _postTodo(text).then(setText(""))}
        />

        <ButtonSubmit onPress={() => _postTodo(text).then(setText(""))}>
          <Icon name="add" color="#fff" size={20}></Icon>
        </ButtonSubmit>
      </Form>
    </InputCreateItemWrapper>
  );
};

const shadow = {
  shadowColor: "#000",
  shadowOffset: {
    width: 0,
    height: 1,
  },
  shadowOpacity: 0.2,
  shadowRadius: 1.41,

  elevation: 2,
};

export default InputCreateItem;
