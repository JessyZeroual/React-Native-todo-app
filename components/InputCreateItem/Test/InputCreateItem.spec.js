import React from "react";
import { TextInput } from "react-native";
import TestRenderer, { act } from "react-test-renderer";
import { TodosContextProvider } from "../../../context/TodosContext";
import InputCreateItem from "../InputCreateItem";

const mockPostTodo = jest.fn(() => Promise.resolve());

describe("InputCreateItem", () => {
  let renderer;
  let input;

  beforeEach(() => {
    act(() => {
      renderer = TestRenderer.create(
        <TodosContextProvider
          value={{
            dispatch: { _postTodo: mockPostTodo },
          }}
        >
          <InputCreateItem />
        </TodosContextProvider>
      );
    });
  });

  describe("When InputCreateItem is mounted", () => {
    it("renders correctly", () => {
      expect(renderer).toMatchSnapshot();
    });
  });

  describe("When we handle input", () => {
    it("change it's value ", async () => {
      input = renderer.root.findByType(TextInput);
      await act(async () => {
        input.props.onChangeText("todo-1");
      });
      expect(input.props.value).toEqual("todo-1");
    });
  });

  describe("When submitting input ", () => {
    it("call function _postTodo ", async () => {
      await act(async () => {
        input.props.onSubmitEditing();
        expect(mockPostTodo).toHaveBeenCalledTimes(1);
        expect(mockPostTodo).toHaveBeenCalledWith(input.props.value);
      });
    });
  });
});
