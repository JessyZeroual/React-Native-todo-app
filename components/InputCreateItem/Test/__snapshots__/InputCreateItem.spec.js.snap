// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputCreateItem When InputCreateItem is mounted renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "borderBottomColor": "#ebebef",
        "borderBottomWidth": 0.5,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#30bddd",
          "paddingBottom": 10,
          "paddingLeft": 10,
          "paddingRight": 10,
          "paddingTop": 10,
        },
      ]
    }
  />
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#fff",
          "borderRadius": 40,
          "display": "flex",
          "flexDirection": "row",
          "height": 40,
          "marginBottom": 10,
          "marginLeft": 20,
          "marginRight": 20,
          "marginTop": -20,
          "paddingLeft": 20,
          "paddingRight": 10,
        },
        Object {
          "elevation": 3,
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 1,
            "width": 0,
          },
          "shadowOpacity": 0.22,
          "shadowRadius": 2.22,
        },
      ]
    }
  >
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      onSubmitEditing={[Function]}
      placeholder="Add item"
      placeholderTextColor="grey"
      rejectResponderTermination={true}
      style={
        Array [
          Object {
            "flexBasis": 0,
            "flexGrow": 1,
            "flexShrink": 1,
          },
        ]
      }
      underlineColorAndroid="transparent"
      value=""
    />
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#30bddd",
            "borderRadius": 30,
            "display": "flex",
            "height": 30,
            "justifyContent": "center",
            "width": 30,
          },
        ]
      }
    >
      <Text />
    </View>
  </View>
</View>
`;
